warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/BulkIngestMapFileLoader.java

```diff
@@ -489,6 +496,26 @@ public void run() {
         log.info("Bulk map file loader shutting down.");
     }
     
+    private void cleanJobDirectoriesOnStartup() throws IOException {
+        Path[] cleanupDirectories = getJobDirectories(new Path(workDir, jobDirPattern + '/' + CLEANUP_FILE_MARKER));
+        for (int i = 0; i < cleanupDirectories.length; i++) {
+            getFileSystem(srcHdfs).delete(cleanupDirectories[i], true);
```

Comment:
Again, you cannot simply assume that we are ready to delete the directories at this point.  We don't know if we successfully moved all the source files yet.

Reply:
I thought that as the point of marking them as job.cleanup. Meaning we know they are successfully moved, therefore they can be deleted?

So it's possible they haven't moved so we would like to run MarkSourceFilesLoaded() again?

Reply:
No, reread your ticket.  The point is that we've seen jobs only partially complete, which leaves source files hanging around.  Notice that you are marking it as job.cleanup before we move the files, not after.  In the past, if the process was interrupted there, the state would still say job.loading and we had no idea at what point the process terminated.  Did it actually finish loading?  Did it finish moving its files and just not get to the final cleanup step?  We had to do manual verification each time and often the job was successful, but didn't finish moving all its files.  Job.cleanup allows us to safely move the files and cleanup automatically.